### ISSUE TITLE
`id-length` does not check identifier in array pattern

### ISSUE THREAD

**Issue by yeonjuan** [eslint/eslint#12832]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** 12.13.0
* **npm Version:** 6.12.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbGVuZ3RoOiBcImVycm9yXCIqLyAgICAgLy8gZGVmYXVsdCBpcyBtaW5pbXVtIDItY2hhcnMgKHsgXCJtaW5cIjogMiB9KVxuLyplc2xpbnQtZW52IGVzNiovXG5jb25zdCBbYV0gPSBmb287Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMSwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6eyJqc3giOnRydWV9fSwicnVsZXMiOnt9LCJlbnYiOnsiYnJvd3NlciI6ZmFsc2V9fX0=)

```js
/*eslint id-length: "error"*/     // default is minimum 2-chars ({ "min": 2 })
/*eslint-env es6*/
const [a] = foo;
```

**What did you expect to happen?**

Warn about short named identifier `a`;

**What actually happened? Please include the actual, raw output from ESLint.**

No error.

**Are you willing to submit a pull request to fix this bug?**

Yes. I already finished the implementation.(locally)

But, I have a question before pull request.

> Examples of correct code for this rule with the default options:
> ...
> var { prop: [x] } = {};
> ...

By the [docs in id-length](https://eslint.org/docs/rules/id-length), there is an example in the correct example section showing the array pattern with the short identifier `x`.
In my understanding, the `x` is an identifier created by array destructuring so it should be checked by this rule.

Am I missing something? Is it intended behavior?

**Comment by kaicataldo** [comment]
Thanks for reporting this! Agreed that this is a bug. The rule already warns for this alternative equivalent:

```js
const a = foo.a;
```
See demo [here](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgaWQtbGVuZ3RoOiBcImVycm9yXCIqLyAgICAgLy8gZGVmYXVsdCBpcyBtaW5pbXVtIDItY2hhcnMgKHsgXCJtaW5cIjogMiB9KVxuLyplc2xpbnQtZW52IGVzNiovXG5cbmNvbnN0IGEgPSBmb28uYTsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjExLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7ImpzeCI6dHJ1ZX19LCJydWxlcyI6e30sImVudiI6eyJicm93c2VyIjpmYWxzZX19fQ==).

**Comment by yeonjuan** [comment]
## To note
After #12881 merged, there are false negative examples in the `id-length.md`:

At Examples of correct code...
```js
/*eslint id-length: "error"*/     // default is minimum 2-chars ({ "min": 2 })
/*eslint-env es6*/
// ...
var { prop: [x] } = {}; // false negative, It should be placed in an incorrect example
// ...
```

This will be fixed in #12839.
